packages/airnode-protocol/test/rrp/requesters/RrpBeaconServerWithManager.sol.js

```diff
@@ -687,189 +687,346 @@ describe('requestBeaconUpdate', function () {
+            encodedTimestamp = now + 1;
+            const secondData = hre.ethers.utils.defaultAbiCoder.encode(
+              ['int256', 'uint256'],
+              [encodedData, encodedTimestamp]
+            );
+            const secondSignature = await airnodeWallet.signMessage(
+              hre.ethers.utils.arrayify(
+                hre.ethers.utils.keccak256(
+                  hre.ethers.utils.solidityPack(['bytes32', 'bytes'], [secondRequestId, secondData])
+                )
+              )
+            );
+            // Fulfill the second beacon update
+            await hre.ethers.provider.send('evm_setNextBlockTimestamp', [now + 2]);
+            await airnodeRrp
+              .connect(sponsorWallet)
+              .fulfill(
+                secondRequestId,
+                airnodeAddress,
                 rrpBeaconServer.address,
                 rrpBeaconServer.interface.getSighash('fulfill'),
-                '0x',
-              ]
-            )
-          );
-          // Request the beacon update
-          await rrpBeaconServer
-            .connect(roles.updateRequester)
-            .requestBeaconUpdate(templateId, roles.sponsor.address, sponsorWalletAddress);
-          // Fulfill with 0 status code
-          const lastBlockTimestamp = (await hre.ethers.provider.getBlock(await hre.ethers.provider.getBlockNumber()))
-            .timestamp;
-          const nextBlockTimestamp = lastBlockTimestamp + 1;
-          await hre.ethers.provider.send('evm_setNextBlockTimestamp', [nextBlockTimestamp]);
-          const decodedData = 123;
-          const data = hre.ethers.utils.defaultAbiCoder.encode(['int256'], [decodedData]);
-          const signature = await airnodeWallet.signMessage(
-            hre.ethers.utils.arrayify(
-              hre.ethers.utils.keccak256(hre.ethers.utils.solidityPack(['bytes32', 'bytes'], [requestId, data]))
-            )
-          );
-          await expect(
-            airnodeRrp
+                secondData,
+                secondSignature,
+                { gasLimit: 500000 }
+              );
+            // Attempt to fulfill the first beacon update
+            await hre.ethers.provider.send('evm_setNextBlockTimestamp', [now + 3]);
+            const staticCallResult = await airnodeRrp
               .connect(sponsorWallet)
-              .fulfill(
-                requestId,
+              .callStatic.fulfill(
+                firstRequestId,
                 airnodeAddress,
                 rrpBeaconServer.address,
                 rrpBeaconServer.interface.getSighash('fulfill'),
-                data,
-                signature,
+                firstData,
+                firstSignature,
                 { gasLimit: 500000 }
-              )
-          )
-            .to.emit(rrpBeaconServer, 'UpdatedBeacon')
-            .withArgs(templateId, requestId, decodedData, nextBlockTimestamp);
+              );
+            expect(staticCallResult.callSuccess).to.equal(false);
```

Comment:
doing a `callStatic` will not result in a state update so we expect the .call to fail . Was this the expected result of this test?

Reply:
`.callStatic` returning false also indicates that a regular transaction attempt would have failed. Maybe we should test both (Airnode testing a fulfillment with a static call will find out that the fulfillment will fail and will not attempt the fulfillment + an already created fulfillment transaction will revert)

Reply:
yeah, makes sense to test both

Reply:
Actually I remember why I'm testing this way. AirnodeRrp `fulfill()` doesn't revert if the target fulfillment function reverts, it only returns a false `callSuccess`, so this is the only/correct way of testing if the RrpBeaconServer `fulfill()` reverts.

Reply:
I added non-static calls just in case https://github.com/api3dao/airnode/pull/695/commits/23f61654bd81ede720e4770ac731c98789fe3c19